ARCHITECTURE.md

```diff
@@ -0,0 +1,100 @@
+## Architecture 
+More insight information about the way `material-tools` works.
+
+#### Quick Links
+- [How does it work](#how-does-it-work)
+- [How to run Angular in NodeJS](#how-to-run-angular-in-nodejs)
+- [Why are some files manually compiled](#why-are-some-type-of-files-compiled-manually)
+- [Under-the-Hood: Understanding the Virtual Context](#under-the-hood-understanding-the-virtual-context)
+- [Under-the-Hood: How to generate a static theme](#under-the-hood-how-to-generate-a-static-theme)
+
+### How does it work
+Material Tools allows developers to build a custom version of [Angular Material 1.x](http://www.github.com/angular/material).
+
+As a developer you could just specifiy the given [components](https://github.com/angular/material/tree/master/src/components) 
+and `material-tools` will run Angular Material in the `NodeJS` environment.
+
+> Read more about [running an Angular module in NodeJS](#how-to-run-angular-in-nodejs)
+
+The tools will then resolve all dependencies of the specified modules and fetch
+all required files.
+
+> Some type of files are going to be manually compiled by tools to have more control about the output.<br/>
+> Read more [why some files need to be compiled manually](#why-are-some-type-of-files-compiled-manually).
+
+At the end all generated output files will be written to a given folder or can be also accessed from NodeJS 
+> Each output file will be also available with `compressed` / `minified` content.
```

Comment:
This is actually in the `README.MD` already and it would be weird to list it again, since this is kind of an extra document to read after the `README.md`.

I'll create a link to the list in the Readme.